### DISCUSSION TITLE
azure-cli is >400 MB on macOS

### DISCUSSION THREAD

**Discussion by olivia-fox** [Homebrew/discussions#2189] [Everyday usage]
Looking at the [long-standing issue](https://github.com/Azure/azure-cli/issues/7387) over at Azure’s repository, does it seem reasonable for the azure-cli package to be over 400 MB...? It seems like the package should only be ~24 MB in size.

```
azure-cli: stable 2.28.0 (bottled), HEAD
Microsoft Azure CLI 2.0
https://docs.microsoft.com/cli/azure/overview
/usr/local/Cellar/azure-cli/2.28.0 (23,179 files, 433.5MB) *
  Poured from bottle on 2021-09-08 at 17:10:13
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/azure-cli.rb
License: MIT
==> Dependencies
Required: openssl@1.1 ✔, python@3.9 ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
```

**Comment by alebcay** [comment]
As mentioned in the upstream issue you linked, [much of the bloat is contained in the Azure SDK](https://github.com/Azure/azure-cli/issues/7387#issuecomment-879529560), which Homebrew pulls in as `resource`s that `azure-cli` depends on (see all of the `resource`s that start with `azure-*` - https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/azure-cli.rb).

Until upstream manages to reduce the size of that dependency, there's nothing Homebrew can do short of ship an `azure-cli` package that is broken or missing functionality.
